src/software/world/ball.cpp

```diff
@@ -1,32 +1,40 @@
 #include "software/world/ball.h"
 
 #include "shared/constants.h"
+#include "software/world/ball_state.h"
 
 Ball::Ball(Point position, Vector velocity, const Timestamp &timestamp,
-           unsigned int history_duration)
-    : positions_(history_duration),
-      velocities_(history_duration),
-      last_update_timestamps(history_duration)
+           unsigned int history_size)
+    : states_(history_size)
```

Comment:
Since we defined that `history_size` should be `> 0`, can we throw an `std::invalid_argument` exception if `history_size` is `<= 0`? Just for extra safety, and easy to test.